libs/common/build.gradle

```diff
@@ -43,3 +43,81 @@ tasks.named('forbiddenApisMain').configure {
   // TODO: Need to decide how we want to handle for forbidden signatures with the changes to server
   replaceSignatureFiles 'jdk-signatures'
 }
+
+// Add support for incubator modules on supported Java versions.
```

Comment:
Note to the reviewers: There were a couple of challenges that led to this overly complicated Gradle config, but I've tried my best to reduce any unwanted changes leaking outside the scope of what's needed.
* We still build with Java 11 at minimum which doesn't support the Vector API. This requires us to set up additional sources for Java 20 (similar to Lucene) and assemble as a multi-release JAR.
* Adding additional unit-tests for Java 20 sources further required creation of new test sources. The runtime classpath depends on the multi-release JAR output so 1) it can load the correct classes based on the Java version, and 2) not run into JarHell problems.
* Use of incubator modules throws a warning. This is hardcoded in OpenJDK with no option to suppress ([link](https://bugs.openjdk.org/browse/JDK-8187591)). As a result, the `-Werror` flag had to be removed for the Java 20 sources.
* Forbidden APIs' check fails on these new sources for two reasons: 1) classes for the Vector API isn't found in the classpath, and 2) some dependency classes defined in the [signatures' file](https://github.com/opensearch-project/OpenSearch/blob/main/buildSrc/src/main/resources/forbidden/opensearch-all-signatures.txt#L52) weren't found in the classpath. These were suppressed using `failOnMissingClasses` and `ignoreSignaturesOfMissingClasses` respectively.

Reply:
@reta Apologies in advance, but you're basically the resident Gradle guru here. Do you mind taking a look at this?

Reply:
Sure, will do shortly :D